packages/Search/examples/viz/tree_visualization.cpp

```diff
@@ -31,28 +33,51 @@ void printPointCloud( View points, std::ostream &os )
 }
 
 template <typename TreeType>
-void viz()
+void viz( std::string const &prefix, std ::string const &infile )
 {
     using DeviceType = typename TreeType::device_type;
     using ExecutionSpace = typename DeviceType::execution_space;
     Kokkos::View<DataTransferKit::Point *, DeviceType> points( "points" );
-    loadPointCloud( "leaf_cloud.txt", points );
+    loadPointCloud( infile, points );
 
     TreeType bvh( points );
 
+    using TreeVisualization =
+        typename DataTransferKit::Details::TreeVisualization<DeviceType>;
+    using TikZVisitor = typename TreeVisualization::TikZVisitor;
+    using GraphvizVisitor = typename TreeVisualization::GraphvizVisitor;
+
+    int const n_neighbors = 10;
+    int const n_queries = bvh.size();
+    Kokkos::View<DataTransferKit::Nearest<DataTransferKit::Point> *, DeviceType>
+        queries( "queries", n_queries );
+    Kokkos::parallel_for( Kokkos::RangePolicy<ExecutionSpace>( 0, n_queries ),
+                          KOKKOS_LAMBDA( int i ) {
+                              queries( i ) = DataTransferKit::nearest(
+                                  points( i ), n_neighbors );
+                          } );
+    Kokkos::fence();
+
+    auto performQueries = [&bvh, &queries]( std::string const &p,
```

Comment:
I would make `queries` an explicit argument of the lambda for clarity. Otherwise, it's weird that `performQueries` call does not have `queries`.

Reply:
I do not really want to spell out the types for `bvh` and `queries` and come up with variable names that do not shadow the original ones.  I could move this out of the function body and template but I actually think it less readable.